lib/rules/keyframe-block-no-duplicate-selectors/index.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const isStandardSyntaxSelector = require('../../utils/isStandardSyntaxSelector');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+
+const ruleName = 'keyframe-block-no-duplicate-selectors';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (selector) => `Unexpected duplicate "${selector}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/list/keyframe-block-no-duplicate-selectors',
+};
+
+/** @type {import('stylelint').Rule} */
+const rule = (primary) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, { actual: primary });
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkAtRules(/^(-(moz|webkit)-)?keyframes$/i, (atRuleKeyframes) => {
+			const selectors = new Set();
+
+			atRuleKeyframes.walkRules((keyframeRule) => {
+				const ruleSelectors = keyframeRule.selectors;
+
+				ruleSelectors.forEach((selector) => {
+					if (!isStandardSyntaxSelector(selector)) {
+						return;
+					}
+
+					const isDuplicate = selectors.has(selector.toLowerCase());
+
+					if (isDuplicate) {
+						report({
+							message: messages.rejected(selector),
+							node: keyframeRule,
+							result,
+							ruleName,
+							word: selector,
+						});
+
+						return;
+					}
+
+					selectors.add(selector.toLowerCase());
```

Comment:
[trivial] It may makes sense a bit to prevent evaluating `selector.toLowerCase()` twice.

```js
const normalizedSelector = selector.toLowerCase();
```

Reply:
Good suggestion, went a bit on autopilot - I'll add that now!